src/relayFeeCalculator/chain-queries/arbitrum.ts

```diff
@@ -19,9 +23,8 @@ export class ArbitrumQueries implements QueryInterface {
   }
 
   async getGasCosts(_tokenSymbol: string): Promise<BigNumberish> {
```

Comment:
There's quite a lot of duplication across these different classes (arbitrum, polygon, etc.). Can we potentially change the QueryInterface they currently implement to be a class they can inherit all these common methods from? I think currently only Polygon has a different implementation of getTokenPrice() and all other ones look exactly the same (just different addresses which can easily be passed in the constructor)

Reply:
I agree - I did notice code-duplication within the chain queries. In this PR I focused primarily on refactoring the duplication within `getGasCosts`

But, I think changing QueryInterface to a class structure (assuming there aren't any frontend/relayer dependencies) would be a great move. As you mentioned, the inheritance would help refactor the code into a more succinct format.

Reply:
It would actually remove all the code for all chains, if I'm not mistaken, except for Polygon, which needs to override getTokenPrice().

Reply:
Yep! And if we pass a `coinGeckoBaseCurrency` variable to the base class, we can use the shared implementation of `super.getTokenPrice()` for the first half of the Polygon `getTokenPrice` custom implementation (further cutting the code reuse).